packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -122,3 +127,28 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
   @override
   String toString() => '$runtimeType("$url", scale: $scale)';
 }
+
+/// Thrown to indicate that loading a network image failed
+///
+class NetworkImageLoadException implements Exception {
```

Comment:
It feel weird to have Exception defined here, and given the information you need, it can be more general for http usage maybe something like HttpException. and I think it should be implemented in [dart sdk](https://github.com/dart-lang/sdk/blob/bbb027aa2ac4ebbb0023227347c3e2d33661b80e/sdk/lib/core/exceptions.dart#L20)
@dnfield do you have any input on this?

Reply:
Can we just use https://api.dartlang.org/stable/2.4.0/dart-io/HttpException-class.html?

Reply:
We don't get the status code in that case, but maybe we coulf file a bug upstream for that if it's important.

Reply:
Agreed, created an issue there. I'll probably do the PR there as well.
https://github.com/dart-lang/sdk/issues/37423

Can we put this with label blocked until the other issue is reolved?